### ISSUE TITLE
[Swift] handle decoding of unknown enum types if optional

### ISSUE THREAD

**Issue by dylan-colaco** [swagger-api/swagger-codegen#7304]
##### Description

If I have a model class, say vehicle which has an enum 'type' which can have two values, 'car' and 'motorbike'. This class is generated with Swagger and instantiated from a JSON response received from my backend(using the codable protocol). Say I ship my App version 1.0 with this.

Later on I decide to add another vehicle type in my backend which will be handled in my app 1.1, 'bus', my app version is was released earlier will not be able to deserialize Vehicle. It will throw an error even if the type is not 'required':
dataCorrupted(Swift.DecodingError.Context(codingPath: ["type"], debugDescription: "Cannot initialize ModelType from invalid String value Something", underlyingError: nil))

##### Swagger-codegen version

swagger-codegen-cli-2.3.0-20171207.164637-326

##### Suggest a fix/enhancement

It would be much nicer to have an option to deserialize optional unknown enum types to nil(like the option READ_UNKNOWN_ENUM_VALUES_AS_NULL in Jackson for Java) instead of throwing an error. This can be done by doing:
```swift
 type = try? container.decodeIfPresent(ModelType.self, forKey: "type")
```
instead of
```  swift
type = try container.decodeIfPresent(ModelType.self, forKey: "type")
```
in the generated method of the class containing the enum
```swift
    public required init(from decoder: Decoder) throws 
```

**Comment by Hoppsen** [comment]
Hey @dylan-colaco,

have you found a solution for this issue? I'm currently struggling with the exact same issue. I have a version out in the app store and with the new enum type this version would not get any results.

Cheers,
Hoppsen

**Comment by dylan-colaco** [comment]
No, I don't. I haven't been working on this, but what I intended to do is have a post processing script to change the _try container.decodeIfPresent_ to _try? container.decodeIfPresent_ in the generated swift files, till the swagger codegen developers fix this. That way the nil value can be gracefully handled
